Mathlib/GroupTheory/GroupAction/Basic.lean

```diff
@@ -102,7 +102,7 @@ end MulAction
 /-- `smul` by a `k : M` over a ring is injective, if `k` is not a zero divisor.
 The general theory of such `k` is elaborated by `IsSMulRegular`.
 The typeclass that restricts all terms of `M` to have this property is `NoZeroSMulDivisors`. -/
-theorem smul_cancel_of_non_zero_divisor {M R : Type*} [Monoid M] [NonUnitalNonAssocRing R]
+theorem smul_cancel_of_non_zero_divisor {M R : Type*} [Monoid M] [AddGroup R]
```

Comment:
Please fix the docstring and rename `R` to `E` (often used for vector spaces in Mathlib) or `G` (often used for groups, both additive and multiplicative).